internal/completions/client/openai/openai.go

```diff
@@ -213,3 +310,11 @@ type openaiResponse struct {
 	Model   string         `json:"model"`
 	Choices []openaiChoice `json:"choices"`
 }
+
+func getPrompt(messages []types.Message) (string, error) {
+	if len(messages) != 1 {
+		return "", errors.New("Expected to receive exactly one message with the prompt")
+	}
```

Comment:
If we were to see this error at runtime, it might be useful to include how many messages were sent? e.g. could it be `messages` was `nil`? Or somehow we are sending 2+?

Also, minor nitpick, per Go idioms:
> Idiomatic error messages in Go should start with a lowercase letter and should contain no trailing punctuation.
> https://docs.sourcegraph.com/dev/background-information/languages/go_errors
 

```suggestion
	if l := len(messages); l != 1 {
		return "", errors.Errof("expected to receive exactly one message with the prompt (got %d)", l)
	}
```